sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseDecoder.java

```diff
@@ -104,11 +104,11 @@ public HttpResponse getSourceResponse() {
          *
          * @return publisher that emits entity instance representing decoded http headers
          */
-        public Mono<Object> getDecodedHeaders() {
+        public Object getDecodedHeaders() {
```

Comment:
What was the reason we lazily decoded headers before?

Reply:
Nope, I'm not completely certain why this was reactive before.